src/machine/machinectl.c

```diff
@@ -1412,6 +1413,119 @@ static int shell_machine(int argc, char *argv[], void *userdata) {
         return process_forward(event, &forward, master, 0, machine);
 }
 
+static int get_settings_path(const char *name, char **ret_path) {
+        _cleanup_free_ char *file_alloc = NULL;
+        const char *file;
+
+        assert(name);
+        assert(ret_path);
+
+        if (!endswith(name, ".nspawn")) {
+                file_alloc = strjoin(name, ".nspawn");
+                if (!file_alloc)
+                        return log_oom();
```

Comment:
Please make each function _always_ log failure or silently (except for log_debug) return negative errno.

This is a kind of library like function, hence `return -ENOMEM;` should be sufficient.